lib/built-in-rules-index.js

```diff
@@ -0,0 +1,275 @@
+/**
+ * @fileoverview Collects the built-in rules into a map structure so that they can be imported all at once and without
+ * using the file-system directly.
+ * @author Peter (Somogyvari) Metz
+ */
+
+"use strict";
```

Comment:
Can you add a test [here](https://github.com/eslint/eslint/blob/04450bb7ed20f2412102538b238119d9764b4dc9/Makefile.js#L940) to ensure that this file actually contains all of the rules? It seems like this file could potentially get out of date when people add new rules.

Reply:
For that matter, do we really need to export the rules explicitly? Or could we use a package to require everything in the directory?

Reply:
That's effectively what we were doing before, but it requires accessing the filesystem at runtime, which is what we were trying to avoid.

Reply:
Ah, right- I guess my hope was that we could access the filesystem, but do so before anyone invokes the `Linter` constructor. If that's not feasible, I don't mind.

Reply:
Added a check, verified it with a manual test: If you drop a key from the index file, the build will fail telling you the rule ID which is missing from the index.